core/src/nucleus/validation/link_entry.rs

```diff
@@ -0,0 +1,60 @@
+use crate::{
+    context::Context,
+    nucleus::{
+        actions::run_validation_callback::run_validation_callback,
+        ribosome::callback::links_utils,
+        validation::{ValidationError, ValidationResult},
+        ZomeFnCall,
+    },
+};
+use holochain_core_types::{
+    cas::content::AddressableContent, entry::Entry, validation::ValidationData,
+};
+use holochain_wasm_utils::api_serialization::validation::LinkValidationArgs;
+use std::sync::Arc;
+
+pub async fn validate_link_entry(
+    entry: Entry,
+    validation_data: ValidationData,
+    context: &Arc<Context>,
+) -> ValidationResult {
+    let address = entry.address().clone();
+    let link = match entry {
+        Entry::LinkAdd(link_add) => link_add,
+        Entry::LinkRemove(link_remove) => link_remove,
+        _ => {
+            return Err(ValidationError::Error(
+                "Could not extract link_add from entry".into(),
+            ));
+        }
+    };
+    let link = link.link().clone();
+    let (base, target) = links_utils::get_link_entries(&link, context).map_err(|_| {
+        ValidationError::UnresolvedDependencies(
+            [link.base().clone(), link.target().clone()].to_vec(),
+        )
```

Comment:
I like that this hooks into the general concept of unresolved dependencies which we can easily use to drive the userland validation case as well :+1: